### PR TITLE
feat: Footer block #11

### LINKED ISSUE #11: Footer block
Every presentation can have its own footer.

- brand logos
- paragraph text
- contact us button
- presentation sources overlay
- SG Creative logo

image

### LINKED ISSUE #50: Footer: Presentation sources
Automatically add a presentation sources button and overlay to footer. The data for the presentation sources could come from an excel sheet for the current presentation (eg. sources.xlsx). The footer block could try to load this data and if it exists, add the `View Presentation Sources` button and then have a click handler that displays an overlay containing all the data from the spreadsheet. Look at `getTheme` and `portfolio` block for examples of loading JSON.

image

image

### PR DESCRIPTION
**Pull request by arumsey**
## Issue

Fix #11 
Fix #50 

## 🔗 Test URLs:

- Before: https://main--sgws-creative--hlxsites.hlx.page/drafts/arumsey/footer
- After: https://footer-block--sgws-creative--hlxsites.hlx.page/bloominbrands/p/1

## 📝 Description:

- consolidated footer loading for presentation and product pages
- added support for sources block within footer fragment

image

image

### CODE REVIEW

File: blocks/footer/footer.css [hunk 1]
```diff
@@ -1,9 +1,202 @@
...
+
+footer .sources {
+  display: none;
+  position: absolute;
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+  background-color: #fff;
+  opacity: 0.9;
```
Comment (tmathern) [inline blocks/footer/footer.css:43]: On my screen it looks less opaque than on the original website - but maybe it's the screen? I'd go with 0.95.

File: blocks/footer/footer.css [hunk 2]
```diff
@@ -1,9 +1,202 @@
...
+}
+
+footer .sources {
+  display: none;
+  position: absolute;
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+  background-color: #fff;
```
Comment (tmathern) [inline blocks/footer/footer.css:42]: Variable?

File: blocks/footer/footer.css [hunk 3]
```diff
@@ -1,9 +1,202 @@
...
+footer h2.animate,
+footer p.animate {
+  animation-delay: 0.25s;
+  animation-duration: 1s;
+  animation-fill-mode: both;
+  animation-name: fade-in-bottom;
+}
+
+.footer a.button:any-link, button {
+  width: 350px;
```
Comment (tmathern) [inline blocks/footer/footer.css:30]: Should be relative to scale too?

File: blocks/footer/footer.css [hunk 4]
```diff
@@ -1,9 +1,202 @@
...
+  border: 2px solid #fff;
```
Comment (tmathern) [inline blocks/footer/footer.css:31]: Variable for the color?

File: blocks/footer/footer.css [hunk 5]
```diff
@@ -1,9 +1,202 @@
...
+  position: absolute;
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+  background-color: #fff;
+  opacity: 0.9;
+  overflow: hidden;
+  padding: 2vw 4vw;
+  font-size: clamp(0.5rem, 0.704vw + 0.236rem, 0.75rem);;
```
Comment (tmathern) [inline blocks/footer/footer.css:46]:
```suggestion
  font-size: clamp(0.5rem, 0.704vw + 0.236rem, 0.75rem);
```

File: blocks/footer/footer.css [hunk 6]
```diff
@@ -1,9 +1,202 @@
...
+  flex-flow: row nowrap;
+  gap: var(--layout-size-s);
+  align-items: center;
+  color: var(--secondary-color);
+}
+
+footer .sources .header .sources-nav button {
+  position: relative;
+  width: 42px;
+  height: 42px;
```
Comment (tmathern) [inline blocks/footer/footer.css:101]: Should be relative to scale too?

File: blocks/footer/footer.css [hunk 7]
```diff
@@ -1,9 +1,202 @@
...
+  display: flex;
+  flex-flow: row nowrap;
+  gap: var(--layout-size-s);
+  align-items: center;
+  color: var(--secondary-color);
+}
+
+footer .sources .header .sources-nav button {
+  position: relative;
+  width: 42px;
```
Comment (tmathern) [inline blocks/footer/footer.css:100]: Should be relative to scale too?

**Review by tmathern — CHANGES_REQUESTED** [review]

File: blocks/footer/footer.css (hunk 4 above)
Comment (arumsey) [reply blocks/footer/footer.css:31]: I think it will always be white (for now at least).

File: blocks/footer/footer.css (hunk 3 above)
Comment (arumsey) [reply blocks/footer/footer.css:30]: Good point. I'll see what can be done.

File: blocks/footer/footer.css (hunk 2 above)
Comment (arumsey) [reply blocks/footer/footer.css:42]: Always will be white.

**Review by allentsliu — APPROVED** [review]
LGTM!

**Review by tmathern — APPROVED** [review]

**Pull request merged**